src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -674,14 +674,17 @@ public Iterator<IndexMetadata> iterator() {
 
             final List<AliasMetadata> filteredAliases = new ArrayList<AliasMetadata>();
 
-            final ImmutableOpenMap<String, AliasMetadata> aliases = indexMetaData.getAliases();
+            final Map<String, AliasMetadata> aliases = new HashMap<>();
```

Comment:
Why did you make this change? Doesn't `indexMetaData.getAliases()` still return an ImmutableOpenMap? Seems like you should avoid making a copy of the map in any case.

Reply:
Hi @andrross, I made this swap because we could not use `keysIt` anymore. We needed `aliases` to be a normal map to then grab and iterator over the keyset.

Reply:
@nknize What's happening here? Pretty sure making a copy of the the map is the exact opposite of your intention with this change.

Reply:
This was the issue that arose when the change was made to core: https://github.com/opensearch-project/security/issues/2714. I just made changes to get things Green. Sorry if this was not the intended implementation...

Reply:
Sorry @scrawfor99! Definitely not trying to blame you here! I don't understand why `keysIt` doesn't work anymore, and I don't think the intent with @nknize's change is to force downstream dependencies to copy data structures when doing a simple iteration.

Reply:
No copies unless you need to make changes to the map (even then the core should return an `unmodifiableMap` so you would only be able to change a clone). 

If you just need an iterator over the keys then use `.keySet().iterator()`